atomic_reactor/schemas/workflow_data.json

```diff
@@ -40,8 +40,11 @@
 
     "files": {"type": "object"},
     "image_components": {
-      "type": ["array", "null"],
-      "items": {"$ref": "#/definitions/rpm_info"}
+      "type": ["object", "null"],
+      "rpm_components": {
+          "type": ["array"],
+          "items": {"$ref": "#/definitions/rpm_info"}
+      }
```

Comment:
I don't think this works as intended, the `rpm_components` key has no meaning to jsonschema

It should be something like this
```json
"image_components": {
  "type": ["object", "null"],
  "patternProperties": {
    ".*": {
      "type": ["array"],
      "items": {"$ref": "#/definitions/rpm_info"}
    }
  }
}
```
(and could this be unit-tested?)

Reply:
fixed and added unitests